Il2CppInspector.Common/Reflection/TypeInfo.cs

```diff
@@ -36,19 +36,6 @@ private class TypeArgumentsComparer : EqualityComparer<TypeInfo[]>
         private Dictionary<int, TypeInfo> generatedArrayTypes = new Dictionary<int, TypeInfo>();
         private TypeInfo generatedByRefType;
         private TypeInfo generatedPointerType;
-        // This property exposes all types which have been generated directly from this one.
-        public IEnumerable<TypeInfo> CachedGeneratedTypes {
-            get {
-                IEnumerable<TypeInfo> result = generatedArrayTypes.Values;
-                if (genericTypeInstances != null)
-                    result = result.Concat(genericTypeInstances.Values);
-                if (generatedByRefType != null)
-                    result = result.Append(generatedByRefType);
-                if (generatedPointerType != null)
-                    result = result.Append(generatedPointerType);
-                return result;
-            }
-        }
```

Comment:
Thinking in terms of APIs, we have MakeByRefType, MakePointerType and MakeArrayType for reflecting the types with an ElementType component. That leaves genericTypeInstances. Should we not make that a public property so that if we come across SomeType<T, U, V> we can easily query its property to see all of the instances that have been compiled?

Will genericTypeInstances be fully populated (ie. all the possible MakeGenericType calls run) by the time the model has been created or are these types created when requested by the user?

The property should probably throw an InvalidOperationException if called on a type that isn't a GenericTypeDefinition.

Reply:
AFAIK there's no property in the usual C# reflection API for getting all `genericTypeInstances`. I am not sure the property is useful: when it's obtained, you only get the instances that have been generated so far (it's a cache), rather than all possible generic instances (that is - you are *not* guaranteed that all possible `MakeGenericType` calls have been executed). Many of these instances are generated dynamically when members of referencing types get enumerated.

Reply:
Oh, and it is exposed via `CachedGeneratedTypes`, of course.

Reply:
Okay that makes sense, I concur.